packages/forms/src/UIForm/merge.js

```diff
@@ -0,0 +1,207 @@
+import React, { PropTypes } from 'react';
+import { merge } from 'talend-json-schema-form-core';
+import get from 'lodash/get';
+
+import FieldTemplate from './fields/FieldTemplate';
+
+/* eslint-disable no-param-reassign */
+
+function parseArray(items, key) {
+	if (items.type === 'object') {
+		return parseProperties(items.properties, false, `${key}[]`);
+	}
+	return [getUISchemaFromObject(items, key)];
+}
+
+function getUISchemaFromObject(schema, key) {
+	const ui = { key };
+	if (schema.title) {
+		ui.title = schema.title;
+	}
+	if (schema.description) {
+		ui.description = schema.description;
+	}
+	if (schema.enum && schema.enumNames) {
+		ui.titleMap = schema.enum.map((value, index) => ({ value, name: schema.enumNames[index] }));
+		delete schema.enumNames;
+	}
+	if (schema.type === 'object') {
+		ui.widget = 'fieldset';
+		ui.items = parseProperties(schema.properties, false, key);
+	} else if (schema.type === 'string') {
+		if (!schema.enum) {
+			ui.widget = 'text';
+		}
+		if (schema.format) {
+			ui.type = schema.format;
+			if (schema.format === 'uri') {
+				ui.type = 'url';
+			}
+			if (schema.format === 'data-url') {
+				ui.type = 'file';
+			}
+		}
+	} else if (schema.type === 'array') {
+		ui.widget = 'array';
+		ui.items = parseArray(schema.items, key);
+	} else if (schema.type === 'number') {
+		ui.widget = 'text';
+		ui.type = 'number';
```

Comment:
In uiSchema, type and widget are the same, but for number, `type` is used to set the input type. You can remove ui.widget = 'text' that is only used to get the `<Text>` component, which is also mapped to `number`